src/topography.f90

```diff
@@ -247,46 +248,42 @@ subroutine topography_deseas(this)
 
     do its = 1, 150   ! Only need high number after massive editing session with fjords. Normally 10 or so sweeps works.
       counter = 0
-      sea_num = 1
+      sea_num = 0
 
       ! Get number of seas
       do j = 2, this%nyt - 1
         i = 1
-        jm = j - 1
-        jp = j + 1
         if (sea(i, j) < land  .and. sea(i, j) > 0) then
-          if (sea(i, j) >= sea_num) then
-            sea(i, j) = sea_num
+          if (sea(i, j) > sea_num) then
             sea_num = max(min(sea_num+1, sea(this%nxt, j), sea(i, j-1), sea(i+1, j), sea(i, j+1)), sea_num)
+            sea(i, j) = sea_num
           end if
         end if
 
         do i = 2, this%nxt - 1
-          im = i - 1
-          ip = i + 1
           if (sea(i, j) < land  .and. sea(i, j) > 0) then
-            if (sea(i, j) >= sea_num) then
-              sea(i, j) = sea_num
+            if (sea(i, j) > sea_num) then
               sea_num = max(min(sea_num+1, sea(i-1, j), sea(i, j-1), sea(i+1, j), sea(i, j+1)), sea_num)
+              sea(i, j) = sea_num
             end if
           end if
         end do
 
         i = this%nxt
         if (sea(i, j) < land  .and. sea(i, j) > 0) then
-          if (sea(i, j) >= sea_num) then
-            sea(i, j) = sea_num
+          if (sea(i, j) > sea_num) then
             sea_num = max(min(sea_num+1, sea(i-1, j), sea(i, j-1), sea(1, j), sea(i, j+1)), sea_num)
+            sea(i, j) = sea_num
           end if
         end if
       end do
 
       j = this%nyt
       do i = 2, this%nxt - 1
         if (sea(i, j) < land  .and. sea(i, j) > 0) then
-          if (sea(i,j) >= sea_num) then
-            sea(i,j) = sea_num
+          if (sea(i,j) > sea_num) then
             sea_num = max(min(sea_num+1, sea(i-1, j), sea(i, j-1), sea(i+1, j)), sea_num)
```

Comment:
should points on the other side of the tripole seam be taken into account?

Reply:
In theory I would say yes, but in practice I don't think it's  an issue. My understanding is that, in the worst case scenario, some seas might get split into two if they intersect the seam, but that's of no practical consequence. The important thing is that the global ocean will be correctly numbered everywhere, and that should be fine.